geode-core/src/main/java/org/apache/geode/cache/query/security/JavaBeanAccessorMethodAuthorizer.java

```diff
@@ -0,0 +1,173 @@
+ * stored are immutable.
+ * <p/>
+ *
+ * @see org.apache.geode.cache.Cache
+ * @see org.apache.geode.cache.query.security.MethodInvocationAuthorizer
+ * @see org.apache.geode.cache.query.security.RestrictedMethodAuthorizer
+ */
+
+public final class JavaBeanAccessorMethodAuthorizer implements MethodInvocationAuthorizer {
+  static final String NULL_PACKAGE_MESSAGE =
+      "A set of allowed packages should be provided to configure the authorizer.";
+  static final String NULL_AUTHORIZER_MESSAGE =
+      "RestrictedMethodAuthorizer should be provided to create this authorizer.";
+  static final String NULL_CACHE_MESSAGE = "Cache should be provided to configure the authorizer.";
+  static final String GEODE_BASE_PACKAGE = "org.apache.geode";
+  private final RestrictedMethodAuthorizer restrictedMethodAuthorizer;
+  private final Set<String> allowedPackages;
+
+  /**
+   * Creates a {@code JavaBeanAccessorMethodAuthorizer} object and initializes it so it can be
+   * safely used in a multi-threaded environment.
+   * <p/>
+   *
+   * Applications can use this constructor as part of the initialization for custom authorizers
+   * (see {@link Declarable#initialize(Cache, Properties)}), when using a declarative approach.
+   *
+   * @param cache the {@code Cache} instance that owns this authorizer, required in order to
+   *        configure the default {@link RestrictedMethodAuthorizer}.
+   * @param allowedPackages the packages containing classes for which 'is' and 'get' methods will
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(Cache cache, Set<String> allowedPackages) {
+    Objects.requireNonNull(cache, NULL_CACHE_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = new RestrictedMethodAuthorizer(cache);
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Creates a {@code JavaBeanAccessorMethodAuthorizer} object and initializes it so it can be
+   * safely used in a multi-threaded environment.
+   * <p/>
+   *
+   * @param restrictedMethodAuthorizer the default {@code RestrictedMethodAuthorizer} to use.
+   * @param allowedPackages the packages containing classes for which 'is' and 'get' methods will
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(RestrictedMethodAuthorizer restrictedMethodAuthorizer,
+      Set<String> allowedPackages) {
+    Objects.requireNonNull(restrictedMethodAuthorizer, NULL_AUTHORIZER_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = restrictedMethodAuthorizer;
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Executes the authorization logic to determine whether the {@code method} is allowed to be
+   * executed on the {@code target} object instance.
+   * <p/>
+   *
```

Comment:
I'd add the following paragraph to the javadoc, mainly to keep consistency with the other authorizers and to make sure the user is aware of this fact:

```
* If the {@code target} object is an instance of {@link Region}, this methods also ensures that
* the user has the {@code DATA:READ} permission granted for the target {@link Region}.
```